File: simple_poker_hands_kotlin/src/main/kotlin/Cards.kt
data class Card(
    val strength: CardStrength,
    val suit: CardSuit,
)

enum class CardSuit {
    SPADE,
    HEART,
    DIAMOND,
    CLUB,
}

enum class CardStrength {
    TWO,
    THREE,
    FOUR,
    FIVE,
    SIX,
    SEVEN,
    EIGHT,
    NINE,
    TEN,
    JACK,
    QUEEN,
    KING,
    ACE,
}

File: simple_poker_hands_rust/src/cards.rs
#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
pub struct Card {
    pub strength: CardStrength,
    pub suit: CardSuit,
}
#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
pub enum CardSuit {
    Spade,
    Heart,
    Diamond,
    Club,
}
#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug)]
pub enum CardStrength {
    Two = 2,
    Three = 3,
    Four = 4,
    Five  = 5,
    Six = 6,
    Seven = 7,
    Eight = 8,
    Nine = 9,
    Ten = 10,
    Jack = 11,
    Queen = 12,
    King = 13,
    Ace = 14,
}
